### PR TITLE
add flake8 checks for pytest scripts

### PR DESCRIPTION
**Pull request by sshedi**
Signed-off-by: Shreenidhi Shedi <sshedi@vmware.com>

### CODE REVIEW

File: python/setup.py.in [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
+# Copyright (C) 2019-2022 VMware, Inc. All Rights Reserved.
 #
 # Licensed under the GNU General Public License v2 (the "License");
 # you may not use this file except in compliance with the License. The terms
 # of the License are located in the COPYING file of this distribution.
 #
 
-import os
-from distutils.core import setup
+from setuptools import setup
```
Comment (sshedi) [inline python/setup.py.in:9]: This is a significant change, distutils is deprecated and will be removed from python-3.12

File: pytests/conftest.py [hunk 2]
```diff
@@ -17,17 +17,17 @@
 import shutil
 import pytest
 import atexit
+import platform
 import requests
 import subprocess
 import configparser
-import distutils.spawn
```
Comment (sshedi) [inline pytests/conftest.py:23]: This is a significant change, distutils is deprecated and will be removed from python-3.12

File: pytests/conftest.py [hunk 3]
```diff
@@ -101,36 +101,44 @@ def __init__(self, cli_args=None):
...
-        return version_parts[0] * 1000 + version_parts[1] * 100 + version_parts[2]
+        version = version_parts[0] * 1000
+        version += version_parts[1] * 100
+        version += version_parts[2]
+        return version
 
     def check_valgrind(self):
         self.config['valgrind_enabled'] = False
-        valgrind = distutils.spawn.find_executable('valgrind')
+        valgrind = shutil.which('valgrind')
```
Comment (sshedi) [inline pytests/conftest.py:118]: let's use which instead of find_executable

File: pytests/tests/test_tdnf_python.py [hunk 4]
```diff
@@ -93,64 +94,64 @@ def test_tdnf_python_install_single(utils):
     pkgs = [sglpkgname]
 
     tdnf_py_erase(pkgs=pkgs)
-    assert (utils.check_package(sglpkgname) == False)
+    assert (utils.check_package(sglpkgname) is False)
```
Comment (oliverkurth) [inline pytests/tests/test_tdnf_python.py]: use `not utils.check_package(sglpkgname)`, same for all the others. There is no need for comparisons.

File: pytests/conftest.py [hunk 5]
```diff
@@ -101,36 +101,44 @@ def __init__(self, cli_args=None):
...
         # check execution environment and enable valgrind if suitable
         self.check_valgrind()
 
     def version_str_to_int(self, version):
         version_parts = version.split('.')
-        return version_parts[0] * 1000 + version_parts[1] * 100 + version_parts[2]
+        version = version_parts[0] * 1000
+        version += version_parts[1] * 100
+        version += version_parts[2]
+        return version
```
Comment (oliverkurth) [inline pytests/conftest.py]: Why are you doing this? I don't think it makes it any more readable.

File: pytests/conftest.py [hunk 6]
```diff
@@ -101,36 +101,44 @@ def __init__(self, cli_args=None):
...
             self.config['valgrind_disabled_reason'] = 'valgrind not found'
             return
 
-        stream = os.popen(valgrind + ' --version') #nosec
+        stream = os.popen(valgrind + ' --version')  # nosec
         valgrind_version = stream.read()
         if not valgrind_version:
-            self.config['valgrind_disabled_reason'] = 'Unable to ascertain valgrind version'
+            reason = 'Unable to ascertain valgrind version'
+            self.config['valgrind_disabled_reason'] = reason
```
Comment (oliverkurth) [inline pytests/conftest.py]: Why? This just adds an extra step.

File: pytests/conftest.py [hunk 7]
```diff
@@ -101,36 +101,44 @@ def __init__(self, cli_args=None):
...
             return
 
         # valgrind versions less than minimum required might not support
         # all the options specified or might have known errors.
-        current_version = self.version_str_to_int(valgrind_version.replace('valgrind-', ''))
-        minimum_version = self.version_str_to_int(self.config['valgrind_minimum_version'])
+        current_version = valgrind_version.replace('valgrind-', '')
+        current_version = self.version_str_to_int(current_version)
+        minimum_version = self.config['valgrind_minimum_version']
+        minimum_version = self.version_str_to_int(minimum_version)
```
Comment (oliverkurth) [inline pytests/conftest.py]: You add extra steps here. This is not an improvement.

File: pytests/tests/test_cache.py [hunk 8]
```diff
@@ -48,9 +52,10 @@ def test_install_without_cache(utils):
     if utils.check_package(pkgname):
         utils.erase_package(pkgname)
 
-    utils.run([ 'tdnf', 'install', '-y', '--nogpgcheck', pkgname ])
+    utils.run(['tdnf', 'install', '-y', '--nogpgcheck', pkgname])
+
+    assert(check_package_in_cache(utils, pkgname) is False)
```
Comment (oliverkurth) [inline pytests/tests/test_cache.py]: use `not check_...`

File: pytests/tests/test_cache.py [hunk 9]
```diff
@@ -60,45 +65,49 @@ def test_install_with_cache(utils):
     if utils.check_package(pkgname):
         utils.erase_package(pkgname)
 
-    utils.run([ 'tdnf', 'install', '-y', '--nogpgcheck', pkgname ])
+    utils.run(['tdnf', 'install', '-y', '--nogpgcheck', pkgname])
+
+    assert(check_package_in_cache(utils, pkgname) is True)
```
Comment (oliverkurth) [inline pytests/tests/test_cache.py]: Just do `assert(check_package_in_cache(utils, pkgname))`

File: pytests/tests/test_cache.py [hunk 10]
```diff
@@ -60,45 +65,49 @@ def test_install_with_cache(utils):
...
+    utils.run(['sed', '-i', '$ a keepcache=false', tdnf_config])
 
     pkgname = utils.config["sglversion_pkgname"]
     if utils.check_package(pkgname):
         utils.erase_package(pkgname)
 
-    utils.run([ 'tdnf', 'install', '-y', '--nogpgcheck', pkgname ])
+    utils.run(['tdnf', 'install', '-y', '--nogpgcheck', pkgname])
+
+    assert(check_package_in_cache(utils, pkgname) is False)
```
Comment (oliverkurth) [inline pytests/tests/test_cache.py]: use `not`.

File: pytests/tests/test_check_update.py [hunk 11]
```diff
@@ -1,44 +1,49 @@
...
+    ret = utils.run(['tdnf', 'check-update', 'abcd', '1234'])
     assert (ret['retval'] == 0)
 
+
 def test_check_update_multi_version_package(utils):
     package = utils.config["mulversion_pkgname"] + '-' + utils.config["mulversion_lower"]
-    ret = utils.run([ 'tdnf', 'install', '-y', '--nogpgcheck', package ])
-    assert (utils.check_package(utils.config["mulversion_pkgname"]) == True)
+    ret = utils.run(['tdnf', 'install', '-y', '--nogpgcheck', package])
+    assert (utils.check_package(utils.config["mulversion_pkgname"]) is True)
```
Comment (oliverkurth) [inline pytests/tests/test_check_update.py]: No need for `is True`.

File: pytests/tests/test_config.py [hunk 12]
```diff
@@ -1,57 +1,62 @@
...
-    utils.run([ 'sed', '-i', '/repodir/d', '/tmp/myrepo/mytdnf.conf' ])
-    utils.run([ 'sed', '-i', '$ a repodir=/tmp/myrepo', '/tmp/myrepo/mytdnf.conf' ])
-    utils.run([ 'cp', tdnf_repo, '/tmp/myrepo/photon.repo' ])
-    utils.run([ 'sed', '-i', 's/enabled=0/enabled=1/g', '/tmp/myrepo/photon.repo' ])
+    utils.run(['mkdir', '-p', '/tmp/myrepo'])
+    utils.run(['cp', tdnf_conf, '/tmp/myrepo/mytdnf.conf'])
+    utils.run(['sed', '-i', '/repodir/d', '/tmp/myrepo/mytdnf.conf'])
+    utils.run(['sed', '-i', '$ a repodir=/tmp/myrepo', '/tmp/myrepo/mytdnf.conf'])
+    utils.run(['cp', tdnf_repo, '/tmp/myrepo/photon.repo'])
+    utils.run(['sed', '-i', 's/enabled=0/enabled=1/g', '/tmp/myrepo/photon.repo'])
```
Comment (oliverkurth) [inline pytests/tests/test_config.py:22]: Not a comment for this review directly, but I think some time we should remove the shell calls calls and use python instead.

File: pytests/tests/test_tdnf_python.py (hunk 4 above)
Comment (sshedi) [reply pytests/tests/test_tdnf_python.py]: I think it's okay to use this and gradually change this when there is time and patience. This is being done in lot of places.

File: pytests/conftest.py (hunk 5 above)
Comment (sshedi) [reply pytests/conftest.py]: I will revert this change. At first I was trying to keep the line length less than 80 :)

File: pytests/conftest.py (hunk 7 above)
Comment (sshedi) [reply pytests/conftest.py]: Same here, miserable effort towards keeping line under 80 columns.

File: pytests/tests/test_config.py (hunk 12 above)
Comment (sshedi) [reply pytests/tests/test_config.py:22]: Agree but this is are simple test cases. Keeping them simple makes life easy. We need to write a bunch of functions to achieve this task but not a big deal TBF.

File: pytests/tests/test_tdnf_python.py (hunk 4 above)
Comment (oliverkurth) [reply pytests/tests/test_tdnf_python.py]: Could probably done with a simple `sed` script. But does not have to be this change.

File: pytests/conftest.py [hunk 13]
```diff
@@ -52,7 +52,7 @@ def dump_cert(fn, cert):
         cert.get_subject().C = 'IN'
         cert.get_subject().ST = 'Karnataka'
         cert.get_subject().L = 'Bangalore'
-        cert.get_subject().O = 'VMware'
+        cert.get_subject().O = 'VMware'  # noqa: E741
```
Comment (oliverkurth) [inline pytests/conftest.py:55]: What is this?

File: pytests/conftest.py (hunk 7 above)
Comment (oliverkurth) [reply pytests/conftest.py]: It's still there though. Please revert. You can split a command line over multiple lines without extra steps.

File: pytests/conftest.py (hunk 13 above)
Comment (sshedi) [reply pytests/conftest.py:55]: It's a simple way to escape a warning from flake8. `noqa` - No Questions Asked.

**Review by oliverkurth — COMMENTED** [review]
Overall looks good.

File: pytests/conftest.py (hunk 13 above)
Comment (oliverkurth) [reply pytests/conftest.py:55]: I see. What's the warning though? Do they object to the "VMware" spelling?

File: pytests/tests/test_tdnf_python.py (hunk 4 above)
Comment (sshedi) [reply pytests/tests/test_tdnf_python.py]: I have replaced them all.

File: pytests/conftest.py (hunk 13 above)
Comment (sshedi) [reply pytests/conftest.py:55]: Some weird one, `pytests/conftest.py:55:28: E741 ambiguous variable name 'O'`

**Pull request merged**